doc/source/ray-observability/user-guides/debug-apps/general-debugging.rst

```diff
@@ -1,56 +1,49 @@
 .. _observability-general-debugging:
 
-General Debugging
+Common Cluster Issues
 =======================
 
-Distributed applications are more powerful yet complicated than non-distributed ones. Some of Ray's behavior might catch
-users off guard while there may be sound arguments for these design choices.
+Distributed applications offer great power but also increased complexity. 
+Some of Ray's behaviors may initially surprise users, but these design choices serve important purposes in distributed computing environments.
 
-This page lists some common issues users may run into. In particular, users think of Ray as running on their local machine, and
-while this is sometimes true, this leads to a lot of issues.
+This document outlines common issues encountered when running Ray in a cluster, highlighting key differences compared to running Ray locally.
 
-Environment variables are not passed from the Driver process to Worker processes
+Environment variables aren't passed from the Driver process to Worker processes
 ---------------------------------------------------------------------------------
 
-**Issue**: If you set an environment variable at the command line (where you run your Driver), it is not passed to all the Workers running in the Cluster
-if the Cluster was started previously.
+**Issue:** When you set an environment variable at the command line that launches your Driver, it isn't propagated to all the Worker processes if the cluster has already started.
 
-**Example**: If you have a file ``baz.py`` in the directory you are running Ray in, and you run the following command:
+**Example:** Suppose you have a file ``baz.py`` in the directory where you run Ray, and you execute the following command:
 
 .. literalinclude:: /ray-observability/doc_code/gotchas.py
-  :language: python
-  :start-after: __env_var_start__
-  :end-before: __env_var_end__
+   :language: python
+   :start-after: __env_var_start__
+   :end-before: __env_var_end__
 
-**Expected behavior**: Most people would expect (as if it was a single process on a single machine) that the environment variables would be the same in all Workers. It won’t be.
-
-**Fix**: Use Runtime Environments to pass environment variables explicity.
-If you call ``ray.init(runtime_env=...)``,
-then the Workers will have the environment variable set.
+**Expected behavior:** Users may expect that setting environment variables on the Driver sends them to all Worker processes as if running on a single machine, but it doesn't.
 
+**Fix:** Enable Runtime Environments to explicitly pass environment variables. When you call ``ray.init(runtime_env=...)``, it sends the specified environment variables to the Workers.
+Alternatively, you can set the environment variable on the Driver before spawning the Workers.
```

Comment:
my understanding: if I have a running cluster, and I set an env var on the head node, that it does not propagate to the existing workers

however, at the time a new worker node is spawned, it inherits the snapshot of environment variables that are set on the head node at that moment

Reply:
nvm, I tested on a cluster and see that I was wrong. I've updated the docs